From 39853913760703f9113e0535a6a4de84d1c5f1d0 Mon Sep 17 00:00:00 2001
From: Guillaume Betous <guillaumex.betous@intel.com>
Date: Thu, 25 Sep 2014 13:17:48 +0200
Subject: [PATCH 1/3] OTA cleanup : declare OSIP as a new partition type

Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-1988
Change-Id: Ia5033a9c63f0cda5072a46248967251fd4e4b2b6
Signed-off-by: Guillaume Betous <guillaumex.betous@intel.com>
Category: feature differentiation
Domain: PUPDR-Recovery
Origin: internal
Upstream-Candidate: no, proprietary
---
 tools/releasetools/common.py          |    2 +-
 tools/releasetools/edify_generator.py |    4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index 815c76c..02dce2b 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -1096,7 +1096,7 @@ DataImage = blockimgdiff.DataImage
 # map recovery.fstab's fs_types to mount/format "partition types"
 PARTITION_TYPES = { "yaffs2": "MTD", "mtd": "MTD",
                     "ext4": "EMMC", "emmc": "EMMC",
-                    "f2fs": "EMMC" }
+                    "f2fs": "EMMC", "osip" : "OSIP" }
 
 def GetTypeAndDevice(mount_point, info):
   fstab = info["fstab"]
diff --git a/tools/releasetools/edify_generator.py b/tools/releasetools/edify_generator.py
index 2bd071d..822eea5 100644
--- a/tools/releasetools/edify_generator.py
+++ b/tools/releasetools/edify_generator.py
@@ -258,6 +258,10 @@ class EdifyGenerator(object):
         self.script.append(
             'write_raw_image(package_extract_file("%(fn)s"), "%(device)s");'
             % args)
+      elif partition_type == "OSIP":
+        self.script.append(
+            'write_osip_image(package_extract_file("%(fn)s"), "%(device)s");'
+            % args)
       elif partition_type == "EMMC":
         if mapfn:
           args["map"] = mapfn
-- 
1.7.9.5

